reporters/ReportVectorMigration.cpp

```diff
@@ -125,6 +145,42 @@ GetReportInstantiator( Kernel::instantiator_function_t* pif )
             {
                 throw IncoherentConfigurationException( __FILE__, __LINE__, __FUNCTION__, "Start_Day", m_StartDay, "End_Day", m_EndDay );
             }
+            for (auto state : m_MustBeInState)
+            {
+                switch (state)
+                {
+                    case VectorStateEnum::STATE_EGG:
+                    {
+                        throw InvalidInputDataException(__FILE__, __LINE__, __FUNCTION__,
+                            "'STATE_EGG' is not a valid state for ReportVectorMigration.\n");
+                    }
+                    case VectorStateEnum::STATE_LARVA:
+                    {
+                        throw InvalidInputDataException(__FILE__, __LINE__, __FUNCTION__,
+                            "'STATE_LARVA' is not a valid state for ReportVectorMigration.\n");
+                    }
+                    case VectorStateEnum::STATE_IMMATURE:
+                    {
+                        throw InvalidInputDataException(__FILE__, __LINE__, __FUNCTION__,
+                            "'STATE_IMMATURE' is not a valid state for ReportVectorMigration.\n");
+                    }
+                    case VectorStateEnum::STATE_MALE:
+                    case VectorStateEnum::STATE_ADULT:
+                    case VectorStateEnum::STATE_INFECTIOUS:
+                    case VectorStateEnum::STATE_INFECTED:
+                        break;
+                }
+            }
+            if (!m_FilenameSuffix.empty())
+            {
+                IdmString tmp_report_name = DEFAULT_NAME;
+                std::vector<IdmString> parts = tmp_report_name.split('.');
+                release_assert(parts.size() == 2);
+                std::string output_fn = parts[0];
+                output_fn += "_" + m_FilenameSuffix + ".csv";
+                SetReportName(output_fn);
```

Comment:
We should standardized this.

Reply:
I can use GetNewReportName for this, actually

Reply:
I got the code mostly from ReportFilter, but I don't think we can add the ReportFilter to ReportVectorMigration because it has so much human stuff in there? We could create a "baseReportFilter" class that's not human-oriented and then we can use that for start/stop, nodes, report name for the ReportFilter and ReportVectorMigration?

Reply:
or I guess add the report name functionality to the baseTextReport?

Reply:
How about making task/ticket for you to look at this after vacation?  It should result in a refactor that has zero impact on user but reduces code reuse via copy-paste.

Reply:
https://github.com/InstituteforDiseaseModeling/EMOD/issues/29